Fix library to work with API changes after 17 Nov 2015

florianbeer opened the pull request:
- Add new permission scopes
- Require access token for all endpoints
- Remove deprecated endpoints
- Fix parameters for endpoints
- Fix pagination

florianbeer committed:
Add new permission scopes

florianbeer committed:
Require access token for all endpoints

florianbeer committed:
Remove deprecated endpoints

florianbeer committed:
Fix parameters for endpoints

Fix typo

florianbeer committed:
Fix pagination

Commit:
Fix error IN getUserMedia API

florianbeer committed:
Merge pull request #1 from melnikoved/master

Fix error IN getUserMedia API

fabiomlferreira commented:
Please approve this request

filkaris commented:
I agree, I was about to make a PR to add new permission scopes and I saw this.
It looks ok and it's important, please merge this.

SamHecquet committed:
Bring back "xRateLimitRemaining" data

florianbeer committed:
Merge pull request #2 from SamHecquet/master

Bring back "xRateLimitRemaining" data

dgsjm commented on src/Instagram.php:
```diff
@@ -525,16 +548,14 @@ public function pagination($obj, $limit = 0)
             }
 
-            return $this->_makeCall($function, $auth, array('cursor' => $obj->pagination->next_cursor, 'count' => $limit));
+            return $this->_makeCall($function, array('next_max_id' => $obj->pagination->next_max_id, 'count' => $count));
```
The pagination call is made with a parameter named "next_max_id". It seems that it should be named "max_id". For example, see the parameters for the ["/users/self/media/recent/" endpoint](https://www.instagram.com/developer/endpoints/users/#get_users_media_recent_self). Am I missing something?

florianbeer replied:
Pagination seems to work for me, but maybe I didn't use that endpoint yet.
Did you test it out? What is the error message you are getting back from the Twitter API?
If this is indeed an error, please make a pull request with the changed parameter name.
Thank you.

dgsjm replied:
I don't get any error from Instagram, but each pagination call returns the same batch of results. Changing "next_max_id" to "max_id" allowed me to get the next batch of items in the paginated list.

Commit:
Fixes getUserFollows and getUserFollower

Readme states that `getUserFollows()` and `getUserFollower()` accepts an `$id` and `$limit`, however these were never used.  Also, nothing was ever returned.

Commit:
Update Instagram.php

Commit:
Defaults set

florianbeer committed:
Merge pull request #7 from olimortimer/patch-1

Fixes getUserFollows and getUserFollower

angelxmoreno commented:
:shipit:

jobollen committed:
Added Proxy functionaly. So, Added cURL-proxy properties. These properties are used in _makeOAuthCall() and _makeCall().

florianbeer committed:
Merge pull request #9 from jobollen/master

Added Proxy functionaly.
